wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -524,6 +531,34 @@ private void validatePrecondition(String precondition, Boolean isConditionSQL, F
   }
 
   private void validateSQLModeDirectives(FailureCollector collector) {
+
+    String recipe = config.getDirectives();
+
+    registry = SystemDirectiveRegistry.INSTANCE;
+    try {
+      registry.reload("default");
+    } catch (DirectiveLoadException e) {
+      throw new RuntimeException(e);
```

Comment:
Add more descriptive messages to all the errors if re-throwing or alternatively just add to failure collector. Can also merge the catch blocks if the errors are handled the same way: Ex: catch (DirectiveParseException | RecipeException e)